c7n/resources/vpc.py

```diff
@@ -2185,6 +2185,64 @@ class resource_type(query.TypeInfo):
         taggable = False
 
 
+@KeyPair.filter_registry.register('unused')
+class UnusedKeyPairs(Filter):
+    """Filters all ec2 keys that are not in use
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: unused-key-pairs
+                resource: aws.key-pair
+                filters:
+                  - unused
+    """
+    annotation_key = 'c7n:unused_keys'
+    permissions = ('ec2:DescribeKeyPairs',)
+    schema = type_schema('unused')
```

Comment:
so generally we've moved to make these filters parameterized to find used/unused from a single filter via boolean parameter. say notifying all users of used keypairs, to stop using them ;-)

Reply:
Oh man, this would scare me with this one. I still want to prevent deletes if the unused filter is not used. I have a fear of someone using the delete by accident and getting locked out of all compute instances. I'll poke around with iterfilters and see if its still possible with the boolean scenario but am erring on the side of caution.

Reply:
Addressed in 55c6436. I left the name as unused opposed to used so the intention can be clear but the docs will show it can also be used to filter for used using the state boolean.